### PR TITLE
chore: add Sphinx 7.0

### PR DESCRIPTION
**Pull request by kryskool**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ readme = "README.md"
 requires-python = ">=3.7"
 dependencies = [
   "beautifulsoup4",
-  "sphinx >= 5.0,<7.0",
+  "sphinx >= 5.0,<=7.0",
```
Comment (dgw) [inline pyproject.toml:20]: This is not really the right way to allow Sphinx 7.0. Specifying `<=7.0` will _only_ allow version 7.0.0—not 7.0.1, not 7.1.0, or any other future releases in the 7.x series.

```suggestion
  "sphinx >= 5.0,<8.0",
```

**Review by dgw — COMMENTED** [review]
Aside from assuming you have tested furo against Sphinx 7.0, this patch as submitted isn't going to function quite the way one might hope.

File: pyproject.toml (hunk 1 above)
Comment (kryskool) [reply pyproject.toml:20]: Hi @dgw 

Yes i agree, as i see 7.0.0 remove a lot of deprecate functions, but if 7.0.1 add new features, and breaks furo
we need to stay in 7.0.0, that's why i choose this method

Regards,

Comment (dgw) [reply pyproject.toml:20]: Sphinx [follows](https://github.com/sphinx-doc/sphinx/blob/ba7408209e84ee413f240afc20f3c6b484a81f8f/doc/internals/contributing.rst#getting-started) SemVer, so they will (do their best to) [avoid breaking compatibility](https://semver.org/#summary) with anything that interfaces with Sphinx until the next major version, 8.0.0.

7.0.1 would be a patch release with only small bug fixes and other non-breaking stuff. Even 7.1 should be fully backward compatible, unless Sphinx's maintainers make a mistake when choosing what changes to accept for 7.1.

Comment (kryskool) [reply pyproject.toml]: Changes done, thanks for your feedback

**Comment by pradyunsg** [comment]
Whoop, I ended up handling this in https://github.com/pradyunsg/furo/commit/591780b0f01378aef769da40e5d7a917304bd1a5 instead. Thanks for filing this PR @kryskool! ^>^

**Pull request closed without merging**